src/api-response.ts

```diff
@@ -3,15 +3,44 @@ import { MimeDefinition, mimeJson } from "./mime";
 
 export type ApiResponse<A = z.ZodTypeAny> = {
   schema: A;
-  mimeTypes: string[];
+  /** @default application/json */
+  mimeTypes?: string[];
+  /** @default 200 for a positive response, 400 for a negative response */
+  statusCode?: number;
 };
 
-export const createApiResponse = <S extends z.ZodTypeAny>(
+type ApiResponseCreationProps<S extends z.ZodTypeAny> = {
+  schema: S;
+  statusCode?: number;
+} & ({ mimeTypes?: [string, ...string[]] } | { mimeType?: string });
+
+export function createApiResponse<S extends z.ZodTypeAny>(
```

Comment:
perhaps I can plan just getting rid of this function completely, solving all cases inside `Endpoint` methods